virttest/utils_test/qemu.py

```diff
@@ -1842,3 +1843,301 @@ def get_all_memorys(self, vm):
         session = self.get_session(vm)
         output = session.cmd_output_safe(cmd, timeout=90)
         return set([_ for _ in output.splitlines() if _])
+
+
+class UsbDevTest(object):
+
+    """
+    Provide basic test for usb devices
+    """
+
+    def __init__(self, test, params, env):
+        self.params = params
+        self.test = test
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+
+    def check_usb_dev(self, usb_type, info_usb_name,
+                      vp_id, vendor, product):
+
+        """
+        check if the usb device exist in guest
+        """
+        if usb_type == 'usb-storage' or usb_type == 'usb-bot':
+            self.add_udisk()
+        timeout = float(self.params.get("login_timeout", 600))
+        session = self.vm.wait_for_login(timeout=timeout)
+        logging.info("Verify USB device in monitor.")
+        o = self.vm.monitor.info("usb")
+        if isinstance(o, dict):
+            o = o.get("return")
+        if info_usb_name and (info_usb_name not in o):
+            raise exceptions.TestFail("Could not find '%s' device, monitor "
+                                      "returns: \n%s" % (usb_type, o))
+        logging.info("The %s device can find in the monitor." % usb_type)
+        chk_list = []
+        if vp_id:
+            chk_list.append(vp_id)
+        if vendor:
+            chk_list.append(vendor)
+        if product:
+            chk_list.append(product)
+        logging.info("Verify %s device in guest." % usb_type)
+        for item in chk_list:
+            output = session.cmd_output("lsusb -v")
+            devices = re.findall(item, output, re.I | re.M)
+            if not devices:
+                raise exceptions.TestFail("Could not find item '%s' in guest" % item)
+        logging.info("The  %s device was found in guest." % usb_type)
+        return True
+
+    def mkimg(self, image="usb", fmt="qcow2", size="200M"):
+
+        """
+        create image file for drive
+        """
+        self.params['images'] += " %s" % image
+        self.params["image_name_%s" % image] = "images/%s" % image
+        self.params["image_format_%s" % image] = fmt
+        self.params["image_size_%s" % image] = size
+        self.params["remove_image_%s" % image] = "yes"
+        self.params["create_image_%s" % image] = "yes"
+        env_process.process_images(env_process.preprocess_image, self.test, self.params)
+
+    def drive_set(self, image, interface='none', fmt='qcow2',
+                  serial=None, cache=None, snapshot='off'):
+
+        """
+        create and set a drive object
+        """
```

Comment:
yes, but it‘s interface can't satisfy my requirement,so i need to define it by myself.